rust-runtime/aws-smithy-runtime/src/client/orchestrator/auth.rs

```diff
@@ -124,6 +125,11 @@ fn extract_endpoint_auth_scheme_config(
     endpoint: &Endpoint,
     scheme_id: AuthSchemeId,
 ) -> Result<AuthSchemeEndpointConfig<'_>, AuthOrchestrationError> {
+    // TODO(P96049742): Endpoint config doesn't currently have a concept of optional auth or "no auth", so
+    // we are short-circuiting lookup of endpoint auth scheme config if that is the selected scheme.
+    if scheme_id == NO_AUTH_SCHEME_ID {
+        return Ok(AuthSchemeEndpointConfig::from(None));
```

Comment:
```suggestion
        return Ok(AuthSchemeEndpointConfig::empty());
```

just a little clearer on intention I think. Could also consider naming this `from_model()` or `inherit()`

Reply:
Fixed in 62446213fcf942f0f7d6c19825f5c0b7b3f85b06